src/s3/commands/presignedURL.ts

```diff
@@ -38,36 +45,124 @@ export async function presignedURLCommand(
         })
 
         await copyToClipboard(url, 'URL', window, env)
+        vscode.window.showInformationMessage(`A presigned URL has been copied to your clipboard`)
     })
 }
 
-export async function promptTime(fileName: string, window = Window.vscode()): Promise<number> {
-    const timeStr = await window.showInputBox({
-        value: '15',
-        prompt: localize(
-            'AWS.s3.promptTime.prompt',
-            'Specify the time (minutes) until URL will expire for path: {0}',
-            fileName
-        ),
-        placeHolder: localize('AWS.s3.promptTime.placeHolder', 'Defaults to 15 minutes'),
-        validateInput: validateTime,
-    })
+function validateTime(time: string): string | undefined {
+    const number = Number(time)
+
+    if (isNaN(Number(time)) || !Number.isSafeInteger(number) || Number(time) <= 0) {
+        return invalidNumberWarning
+    }
+
+    return undefined
+}
+
+export interface PresignedUrlWizardState {
+    region: string
+    signedUrlParams: SignedUrlRequest
+}
+
+export class PresignedUrlWizard extends Wizard<PresignedUrlWizardState> {
+    constructor(node?: S3FileNode) {
+        super({
+            initState: {
+                region: node?.bucket.region,
+                signedUrlParams: node ? ({ bucketName: node.bucket.name } as SignedUrlRequest) : undefined,
+            },
+        })
+        this.form.region.bindPrompter(() => createRegionPrompter().transform(region => region.id))
+        this.form.signedUrlParams.operation.bindPrompter(() => createOperationPrompter())
+        this.form.signedUrlParams.bucketName.bindPrompter(({ region }) =>
+            createBucketPrompter(assertDefined(region, 'region'))
+        )
+
+        if (node) {
+            this.form.signedUrlParams.key.setDefault(node.file.key)
+            this.form.signedUrlParams.key.bindPrompter(
+                ({ region, signedUrlParams }) =>
+                    createS3FilePrompter(
+                        assertDefined(region, 'region'),
+                        assertDefined(signedUrlParams?.bucketName, 'bucketName'),
+                        assertDefined(signedUrlParams?.operation, 'operation')
+                    ),
+                { showWhen: state => state.signedUrlParams?.operation === 'putObject' }
+            )
```

Comment:
Not sure if this is the behavior we want. If the user explicitly selects a file node I would assume that they want a pre-signed URL for _that_ destination right?

Reply:
For GET that is the case, but for PUT I don't know if that's what we want. It will replace that object in their bucket. They may just want to insert a new object. Asking for the key is safer in my opinion.